pallets/liquidity-pools-gateway/routers/src/lib.rs

```diff
@@ -257,7 +253,10 @@ where
 			true,
 		)?;
 
-		Ok(())
+		Ok(PostDispatchInfo {
+			actual_weight: Some(self.xcm_domain.overall_weight),
```

Comment:
The above call to `transact_through_sovereign` returns a normal `DispatchResult`, meaning that we cannot get the weight of the call. If there is another way of retrieving the weight of that call, please let me know and I will adjust.